src/receive_imf/tests.rs

```diff
@@ -4855,6 +4855,35 @@ async fn test_protected_group_add_remove_member_missing_key() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_protected_group_reply_from_mua() -> Result<()> {
+    let mut tcm = TestContextManager::new();
+    let alice = &tcm.alice().await;
+    let bob = &tcm.bob().await;
+    let fiona = &tcm.fiona().await;
+    mark_as_verified(alice, bob).await;
+    mark_as_verified(alice, fiona).await;
+    mark_as_verified(bob, alice).await;
+    let alice_chat_id = alice
+        .create_group_with_members(ProtectionStatus::Protected, "Group", &[bob, fiona])
+        .await;
+    let sent = alice.send_text(alice_chat_id, "Hello!").await;
+    let bob_msg = bob.recv_msg(&sent).await;
+    bob_msg.chat_id.accept(bob).await?;
+    bob.sql
+        .execute(
+            "UPDATE chats SET protected=0, grpid='' WHERE id=?",
+            (bob_msg.chat_id,),
+        )
+        .await?;
```

Comment:
This is quite a hacky way of simulating a MUA replying - which is fine since it wouldn't be easy to do it less hacky, but can this get a comment explaining why it works and that it's hacky but there is no other simple way?

Probably the less-hacky but a lot more complicated way would be sth like:

```rust
let encrypted_text = r"Subject: ...
[Other headers]

/me replying from MUA";
let encrypted = EncryptHelper::encrypt(encrypted_text); // this part will be a lot harder in reality
receive_imf(
    &alice,
    format!("From: bob@example.net
[Other headers]
{encrypted}"
);
```

Reply:
I like the hackyness :P but agree, that it is better style to give a short comment